app/models/service_orchestration.rb

```diff
@@ -17,27 +17,48 @@ def stack_name=(stname)
   end
 
   def orchestration_stack_status
-    return "check_status_failed", "stack has not been deployed" unless orchestration_stack
+    return ['deploy_failed', "can't find orchestration stack job for the service"] unless orchestration_runner_job
 
-    orchestration_stack.raw_status.normalized_status
-  rescue MiqException::MiqOrchestrationStackNotExistError, MiqException::MiqOrchestrationStatusError => err
-    # naming convention requires status to end with "failed"
-    ["check_status_failed", err.message]
+    return ['deploy_failed', orchestration_runner_job.message] if orchestration_runner_job.status == 'error'
+
+    return ['deploy_active', 'waiting for the orchestration stack status'] unless orchestration_runner_job.orchestration_stack_status
+
+    [orchestration_runner_job.orchestration_stack_status, orchestration_runner_job.orchestration_stack_message]
   end
 
   def deploy_orchestration_stack
-    creation_options = stack_options
-    @orchestration_stack = ManageIQ::Providers::CloudManager::OrchestrationStack.create_stack(
-      orchestration_manager, stack_name, orchestration_template, creation_options
+    job_options = {
```

Comment:
why not place all options together without a merge?

Reply:
```stack_options``` might change ```orchestration_manager``` and ```orchestration_template```.

Reply:
It is hard to understand. In this case, we can do
```
  deploy_stack_options = stack_options
  job_options = {
    :create_options => deploy_stack_options,
    :orchestration_manager_id => orchestration_manager.id,
    ...
  }
```